src/app/qgsmaptoolselectionhandler.cpp

```diff
@@ -192,6 +195,25 @@ void QgsMapToolSelectionHandler::selectFeaturesPressEvent( QgsMapMouseEvent *e )
 
 void QgsMapToolSelectionHandler::selectFeaturesMoveEvent( QgsMapMouseEvent *e )
 {
+
+  if ( mSelectionMode == QgsMapToolSelectionHandler::SelectOnMouseOver )
+  {
+    // This is a (well known, according to google) false positive,
+    // I tried all possible NOLINT placements without success, this
+    // ugly ifdef seems to do the trick with silencing the warning.
+#ifndef __clang_analyzer__
+    mOnMouseMoveDelayTimer = std::make_unique<QTimer>( );
+    mOnMouseMoveDelayTimer->singleShot( 300, mOnMouseMoveDelayTimer.get(), [ = ]
```

Comment:
If I'm intrepreting this correctly, then we ONLY get the identification after 300ms of no mouse movements. If so, I'd suggest a small tweak which is that a timer is started AFTER a move and identify, and it prevents further identifaction till after 300ms has elapsed since the last one. 

So something like this:
- make a mMoveLastCursorPos member, which is always updated when the move event occurs (regardless of timer state). In the move event if the timer isn't already running, then we immediately do the identify and also start the timer. If the timer **is** already running then we update the mMoveLastCursorPos member but don't restart the timer.
- In the setSelectedGeometry method, we clear mMoveLastCursorPos to an empty QgsPointXY
- On timer timeout, if mMoveLastCursorPos is non empty, then do the actual identification

This would help accessibility or for users with styluses, where it's really hard to keep the cursor still for a period of time.